drivers/counter/counter_mcux_rtc.c

```diff
@@ -236,6 +236,16 @@ static int mcux_rtc_init(const struct device *dev)
 	RTC_GetDefaultConfig(&rtc_config);
 	RTC_Init(config->base, &rtc_config);
 
+#if (defined(FSL_FEATURE_RTC_HAS_LPO_ADJUST) && FSL_FEATURE_RTC_HAS_LPO_ADJUST)
+	BUILD_ASSERT(DT_INST_ENUM_IDX(0, clock_source) == 1,
```

Comment:
shouldn't this build assert be outside the guard? if `FSL_FEATURE_RTC_HAS_LPO_ADJUST` is not defined, then this block won't be compiled in. Also the message should be `cannot chose LPO`, which is the dt enum index 1.

What we want is to prevent the user to chose LPO as source clock when is not available in the device.

Reply:
Thanks for reminding, I have updated.